lib/engine/game/base.rb

```diff
@@ -2083,6 +2083,12 @@ def bankruptcy_limit_reached?
 
       def update_tile_lists(tile, old_tile)
         add_extra_tile(tile) if tile.unlimited
+
+        if (hex = tile.hex) && (hex == hex_by_id(hex.id))
```

Comment:
I don't follow the case you are trying to prevent. What circumstances does the tile have a hex, but that hex isn't a valid hex?